### ISSUE TITLE
Error - Provider org.jclouds.aws.s3.AWSS3ProviderMetadata could not be instantiated

### ISSUE THREAD

**Issue by jaehyeon-kim** [lensesio/stream-reactor#834]
# Issue Guidelines

Please review these questions before submitting any issue?

### What version of the Stream Reactor are you reporting this issue for?

kafka-connect-aws-s3-3.0.1-2.5.0

### Are you running the correct version of Kafka/Confluent for the Stream reactor release?

Not sure, I tried on `bitnami/kafka:2.8.1` and MSK Connect (2.7.1).

### Do you have a supported version of the data source/sink .i.e Cassandra 3.0.9?

Not sure

### Have you read the [docs](https://docs.lenses.io/connectors/)?

Yes

### What is the expected behaviour?

S3 sink connector should push data to S3

### What was observed?

Error - `Provider org.jclouds.aws.s3.AWSS3ProviderMetadata could not be instantiated`

```
connect          | [2022-02-18 08:28:04,370] INFO S3ConfigDefBuilder values:
connect          |      connect.s3.aws.access.key = [hidden]
connect          |      connect.s3.aws.auth.mode = Default
connect          |      connect.s3.aws.region =
connect          |      connect.s3.aws.secret.key = [hidden]
connect          |      connect.s3.custom.endpoint = https://s3.ap-southeast-2.amazonaws.com/
connect          |      connect.s3.disable.flush.count = false
connect          |      connect.s3.error.policy = THROW
connect          |      connect.s3.http.max.retries = 5
connect          |      connect.s3.http.retry.interval = 50
connect          |      connect.s3.kcql = INSERT INTO analytics-data-590312749310-ap-southeast-2:cdc-events SELECT * FROM avo.ods.cdc_events PARTITIONBY customer_id,order_id STOREAS `avro` WITH_FLUSH_INTERVAL = 60 WITH_FLUSH_COUNT = 50
connect          |      connect.s3.local.tmp.directory =
connect          |      connect.s3.max.retries = 20
connect          |      connect.s3.retry.interval = 60000
connect          |      connect.s3.vhost.bucket = false
connect          |  (io.lenses.streamreactor.connect.aws.s3.config.S3ConfigDefBuilder:372)
connect          | [2022-02-18 08:28:11,077] ERROR WorkerSinkTask{id=orders-sink-avo-0} Task threw an uncaught and unrecoverable exception. Task is being killed and will not recover until manually restarted (org.apache.kafka.connect.runtime.WorkerTask:190)
connect          | java.lang.IllegalStateException: org.jclouds.providers.ProviderMetadata: Provider org.jclouds.aws.s3.AWSS3ProviderMetadata could not be instantiated
connect          |      at io.lenses.streamreactor.connect.aws.s3.sink.ThrowableEither.toThrowable(ThrowableEither.scala:34)
connect          |      at io.lenses.streamreactor.connect.aws.s3.sink.S3SinkTask.start(S3SinkTask.scala:65)
connect          |      at org.apache.kafka.connect.runtime.WorkerSinkTask.initializeAndStart(WorkerSinkTask.java:309)
connect          |      at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:186)
connect          |      at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:237)
connect          |      at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
connect          |      at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
connect          |      at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
connect          |      at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
connect          |      at java.base/java.lang.Thread.run(Thread.java:829)
connect          | [2022-02-18 08:28:11,080] INFO Metrics scheduler closed (org.apache.kafka.common.metrics.Metrics:659)
connect          | [2022-02-18 08:28:11,081] INFO Closing reporter org.apache.kafka.common.metrics.JmxReporter (org.apache.kafka.common.metrics.Metrics:663)
connect          | [2022-02-18 08:28:11,081] INFO Metrics reporters closed (org.apache.kafka.common.metrics.Metrics:669)
connect          | [2022-02-18 08:28:11,083] INFO App info kafka.consumer for connector-consumer-orders-sink-avo-0 unregistered (org.apache.kafka.common.utils.AppInfoParser:83)
```

### What is your Connect cluster configuration (connect-avro-distributed.properties)?

Combined with JAR files of Kafka Connect Avro Converter - https://www.confluent.io/hub/confluentinc/kafka-connect-avro-converter
(It's not possible to set plugin path in MSK connect so that they are put together - it worked with Debezium).

![image](https://user-images.githubusercontent.com/1023748/154647439-44b1af68-82d4-420b-bee2-9980d67dbc3a.png)

### What is your connector properties configuration (my-connector.properties)?

```
curl -i -X POST -H "Accept:application/json" -H  "Content-Type:application/json" \
  http://localhost:8083/connectors/ \
  -d '{
        "name": "orders-sink-avo",
        "config": {
          "connector.class": "io.lenses.streamreactor.connect.aws.s3.sink.S3SinkConnector",
          "tasks.max": "1",
          "connect.s3.kcql": "INSERT INTO analytics-data-590312749310-ap-southeast-2:cdc-events SELECT * FROM avo.ods.cdc_events PARTITIONBY customer_id,order_id STOREAS `avro` WITH_FLUSH_INTERVAL = 60 WITH_FLUSH_COUNT = 50",
          "aws.region": "ap-southeast-2",
          "aws.custom.endpoint": "https://s3.ap-southeast-2.amazonaws.com/",
          "topics": "avo.ods.cdc_events",
          "key.converter": "io.confluent.connect.avro.AvroConverter",
          "key.converter.schema.registry.url": "http://registry:8080/apis/ccompat/v6",
          "value.converter": "io.confluent.connect.avro.AvroConverter",
          "value.converter.schema.registry.url": "http://registry:8080/apis/ccompat/v6",
          "errors.log.enable": "true"
        }
}'
```

### Please provide full log files (redact and sensitive information)

See above.

**Comment by jaehyeon-kim** [comment]
Tested with 4 connectors in MSK Connect
* Debezium Postgres 1.8.0 - *works*
* Confluent S3 Sink 10.0.5 - *works*
* Lenses S3 3.0.1 - *fails*
* Voluble 0.3.1 - *works*

**Comment by davidsloan** [comment]
We are currently testing a new release of the connector which should resolve this problem if you configure it to use the AWS library (config below).

If you are able to please download the snapshot from here:
https://drive.google.com/drive/folders/1egDmcFv3DQToUQnbUGEqZrgBKcFmQNKO

and add these to the connector config

connect.s3.aws.client=AWS
connect.s3.aws.region=eu-west-1

(obviously change the region to the correct region for your S3 bucket).

I'd be keen to know if this resolves your problem.

**Comment by davidsloan** [comment]
Hi @jaehyeon-kim were you able to get this working with the October 4.0.0 release?

**Comment by davidsloan** [comment]
Closing as this was fixed with the 4.0.0 release and configuration changes.  Let me know if the problem recurs.
